main.tf

```diff
@@ -453,3 +210,402 @@ resource "aws_s3_bucket_ownership_controls" "this" {
     aws_s3_bucket.this
   ]
 }
+
+resource "aws_s3_bucket_accelerate_configuration" "this" {
+  count  = var.acceleration_status != null ? 1 : 0
+  bucket = aws_s3_bucket.this[0].id
+  status = var.acceleration_status
+}
+
+# TODO: This should accept both ACL and Grant, exclusively
+resource "aws_s3_bucket_acl" "this" {
```

Comment:
I think we can accomplish this with this hacky bit of code (my kingdom for an exclusive OR, HashiCorp...):
```hcl
# If ACL and ACP: 0, if ACL or ACP: 1, if ACL nor ACP: 0
count = (var.acl != "null" && var.acl != null) && length(keys(try(jsondecode(var.access_control_policy), var.access_control_policy))) > 0 ? 0 : (var.acl != "null" && var.acl != null) || length(keys(try(jsondecode(var.access_control_policy), var.access_control_policy))) > 0 ? 1 : 0
```

Reply:
From my understanding, [aws_s3_bucket_acl](https://registry.terraform.io/providers/hashicorp%20%20/aws/latest/docs/resources/s3_bucket_acl) still supports both `acl` and `access_control_policy`, just not at the same time. Would just using something like 
```
count = (var.acl != null || var.access_control_policy != "") ? 1 : 0 
```
get the expectedc result?